File: WEB/finly-go-20250101/internal/server/mongodb.go
package server

import (
	"context"
	"finly-go/pkg/logger"

	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
	"go.uber.org/zap"
)

var mongoDb *mongo.Database

func ConnectMongoDB() (*mongo.Client, error) {
	mongoURI := "mongodb://admin:securepassword123@localhost:27017"
	dbName := "finly"

	ctx := context.Background()

	client, err := mongo.Connect(ctx, options.Client().ApplyURI(mongoURI))
	if err != nil {
		logger.Log.Error("Failed to connect to MongoDB", zap.String("uri", mongoURI), zap.Error(err))
		return nil, err
	}

	if err := client.Ping(ctx, nil); err != nil {
		logger.Log.Error("Failed to ping MongoDB", zap.Error(err))
		return nil, err
	}

	mongoDb = client.Database(dbName)

	logger.Log.Info("Successfully connected to MongoDB", zap.String("database", dbName))
	return client, nil
}

type Collections struct {
	UserCollection     *mongo.Collection
	CustomerCollection *mongo.Collection
	InvoiceCollection  *mongo.Collection
}

func GetCollections() *Collections {
	return &Collections{
		UserCollection:     mongoDb.Collection("users"),
		CustomerCollection: mongoDb.Collection("customers"),
		InvoiceCollection:  mongoDb.Collection("invoices"),
	}
}

File: www/taskly-20241228/server/src/libs/mongodb.js
import { MongoClient } from 'mongodb';
import { Config } from './config.js';
import log from './logger.js';

let mongoDb = null;
const url = Config.MONGO.URI;
const dbName = Config.MONGO.DB;
const client = new MongoClient(url);

export const connectMongo = async () => {
  if (mongoDb) {
    log.info('MongoDB connection already established.');
    return mongoDb;
  }

  try {
    await client.connect();
    log.info('MongoDB connected successfully.');
    mongoDb = client.db(dbName);
    return mongoDb;
  } catch (error) {
    log.error('Failed to connect to MongoDB.', error);
    throw error;
  }
};

export const getDb = () => {
  if (!mongoDb) {
    throw new Error('MongoDB not connected.');
  }
  return mongoDb;
};

export const closeMongo = async () => {
  if (mongoDb) {
    await client.close();
    log.info('MongoDB connection closed.');
    mongoDb = null;
  } else {
    log.info('No active MongoDB connection to close.');
  }
};

export const mongoDbInstance = await connectMongo();
